Fix(mobile) avoid fetchining non-ada tokens in portfolio

AbdullayevAbdurashid opened the pull request:
https://emurgo.atlassian.net/browse/YV-618?atlOrigin=eyJpIjoiMDU3YWIyYWFhM2NjNDFjMzg5YzdmOGFjYzRkZTUyMTgiLCJwIjoiaiJ9

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds a `disableNonPrimaryToken` flag to `useGetPortfolioTokenChart` and enables it in `PortfolioTokenChart` to avoid fetching non-primary token data.
> 
> - **Hooks**:
>   - Add `disableNonPrimaryToken` option to `useGetPortfolioTokenChart` and use it to disable the non-primary token query (`enabled` condition updated); spread remaining `queryOptions` into queries.
> - **UI**:
>   - Update `PortfolioTokenChart` to call `useGetPortfolioTokenChart(timeInterval, {disableNonPrimaryToken: true})`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c4005cea70e74943c6b3b73cd572302e219bff1c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

stackchain commented:
hey @AbdullayevAbdurashid i think the changes were on top of the previous branch, please rebase on develop and drop the duplicated commit for QRs.

banklesss commented on mobile/src/features/Portfolio/common/hooks/useGetPortfolioTokenChart.ts:
```diff
@@ -156,10 +156,10 @@ export const useGetPortfolioTokenChart = (
     refetchOnMount: false,
-    enabled: tokenInfo && !isPrimaryToken(tokenInfo.info),
     ...options,
+    enabled: false, // Disable chart fetching for non-ADA tokens
```
maybe 

`enabled: !disableNonAda && tokenInfo && !isPrimaryToken(tokenInfo.info),` ??

banklesss replied:
plus

`useGetPortfolioTokenChart({disableNonAda: true})`

banklesss commented on mobile/src/features/Portfolio/context/PortfolioTokenActivityProvider.tsx:
```diff
@@ -113,7 +113,7 @@ export const PortfolioTokenActivityProvider = ({
 
   const query = useQuery({
-    enabled: state.secondaryTokenIds.length > 0,
+    enabled: false, // Disabled: API only supports primary token (ADA) activity, not secondary tokens
```
why this change? is it necessary?

AbdullayevAbdurashid replied:
I thought this query was causing error at first but it wasnt.Returned to the first state

AbdullayevAbdurashid committed:
fix(mobile): YV-618 - disable chart data fetching for non-ADA tokens

- Add disableNonPrimaryToken flag to useGetPortfolioTokenChart hook
- Enable flag in PortfolioTokenChart component to prevent failing API calls
- Only primary token (ADA) will fetch chart data

banklesss committed:
Merge branch 'develop' into fix(mobile)-YV-618-portfolio-avoid-failing-calls-to-fetch-chart

banklesss committed:
Merge branch 'develop' into fix(mobile)-YV-618-portfolio-avoid-failing-calls-to-fetch-chart

banklesss committed:
Merge branch 'develop' into fix(mobile)-YV-618-portfolio-avoid-failing-calls-to-fetch-chart
